K2Bridge/Visitors/Aggregations/MetricAggregationVisitor.cs

```diff
@@ -108,10 +127,58 @@ public void Visit(PercentileAggregation percentileAggregation)
             var valuesForColumnNames = string.Join(sep, percentileAggregation.Percents.ToList().Select(item => $"{item:0.0}"));
             var valuesForOperator = string.Join(',', percentileAggregation.Percents);
 
+            var metadata = new List<string>();
+            metadata.Add(percentileAggregation.Key);
+            metadata.Add(AggregationsConstants.Percentile);
+            metadata.Add(valuesForColumnNames);
+            metadata.Add(percentileAggregation.Keyed.ToString());
+
             // We don't use EncodeKustoField on this key because it contains a '.' but isn't dynamic
             // Example: ['A%percentile%25.0%50.0%99.0%False']=percentiles_array(fieldA, 25,50,99)']
-            var key = $"['{percentileAggregation.Key}{sep}percentile{sep}{valuesForColumnNames}{sep}{percentileAggregation.Keyed}']";
-            percentileAggregation.KustoQL = $"{key}={KustoQLOperators.PercentilesArray}({EncodeKustoField(percentileAggregation)}, {valuesForOperator})";
+            var keyWithMetadata = $"['{string.Join(AggregationsConstants.MetadataSeparator, metadata)}']";
+            percentileAggregation.KustoQL = $"{keyWithMetadata}={KustoQLOperators.PercentilesArray}({EncodeKustoField(percentileAggregation)}, {valuesForOperator})";
+
+            VisitedMetrics.Add(keyWithMetadata);
```

Comment:
same question.
I guess the answer is NO . but why

Reply:
Same answer :)